pkg/apis/numaflow/v1alpha1/mono_vertex_types.go

```diff
@@ -609,6 +621,8 @@ func (mvs *MonoVertexStatus) IsHealthy() bool {
 	// We check if all the required conditions are true for it to be healthy
 	case MonoVertexPhaseRunning:
 		return mvs.IsReady()
+	case MonoVertexPhasePaused:
```

Comment:
For paused phase, we also need to check `IsReady()` right? because daemon is still there.

Reply:
I kept it consistent with our pipeline health check. 

https://github.com/numaproj/numaflow/blob/b8c61debb784db0eaae31352138fd9b447f9d151/pkg/apis/numaflow/v1alpha1/pipeline_types.go#L782

Do we want to add that now for both?

Reply:
Ideally, we should have check for both to ensure no changes in the conditions